### PR TITLE
docs(headless): documentation for sort criteria

### PR DESCRIPTION
**Pull request by olamothe**
https://coveord.atlassian.net/browse/KIT-327

### CODE REVIEW

**Review by ThibodeauJF — APPROVED** [review]

File: packages/headless/src/features/sort-criteria/criteria.ts [hunk 1]
```diff
@@ -1,22 +1,58 @@
 import {isArray} from '@coveo/bueno';
 
+/**
+ * The available sort order
```
Comment (ThibodeauJF) [inline packages/headless/src/features/sort-criteria/criteria.ts:4]: I believe it takes dots at the end of doc descriptions, here and in other descriptions

File: packages/headless/src/features/sort-criteria/criteria.ts [hunk 2]
```diff
@@ -1,22 +1,58 @@
...
   Ascending = 'ascending',
   Descending = 'descending',
 }
 
+/**
+ * The available criteria that can be used to sort query results
+ */
 export enum SortBy {
+  /**
+   * Use standard index ranking factors (adjacency, TDIDF) and custom ranking expressions (QREs and QRFs) to compute a ranking score for each query result item, and sort the query results by descending score value
```
Comment (ThibodeauJF) [inline packages/headless/src/features/sort-criteria/criteria.ts:16]: Should it be third person? Uses, Sorts, Does not, etc.

File: packages/headless/src/features/sort-criteria/criteria.ts [hunk 3]
```diff
@@ -26,6 +62,10 @@ export type SortCriterion =
   | SortByField
   | SortByNoSort;
 
+/**
+ * Allows to build a sort expression that can be understood and executed by the Coveo platform.
+ * @param {SortCriterion} criterion The criterion to translate to a valid sort query expression.
```
Comment (ThibodeauJF) [inline packages/headless/src/features/sort-criteria/criteria.ts:67]: criterion is repeated

File: packages/headless/src/features/sort-criteria/criteria.ts [hunk 4]
```diff
@@ -48,15 +88,30 @@ export const buildCriterionExpression = (
...
+ * Utility function that builds a valid sort by relevancy criterion.
+ * @returns {SortByRelevancy} The sort by relevancy criterion.
+ */
 export const buildRelevanceSortCriterion = (): SortByRelevancy => ({
   by: SortBy.Relevancy,
 });
 
+/**
+ * Utility function that builds a valid sort by date criterion.
+ * @param {SortOrder} order The order (descending/descending) on which to sort.
```
Comment (ThibodeauJF) [inline packages/headless/src/features/sort-criteria/criteria.ts:101]: order is repeated here and on line 112

Comment (ThibodeauJF) [reply packages/headless/src/features/sort-criteria/criteria.ts:101]: (ascending/descending)

File: packages/headless/src/features/sort-criteria/criteria.ts [hunk 5]
```diff
@@ -66,8 +121,16 @@ export const buildFieldSortCriterion = (
...
+ * Utility function that builds a valid sort by QRE criterion.
+ * @returns {SortByQRE} The sort by QRE criterion.
+ */
 export const buildQueryRankingExpressionSortCriterion = (): SortByQRE => ({
   by: SortBy.QRE,
 });
 
+/**
+ * Utility function that builds a valid no sort criterion.
+ * @returns {SortByNoSort} The sort by no sort criterion.
```
Comment (ThibodeauJF) [inline packages/headless/src/features/sort-criteria/criteria.ts:134]: I feel like this doesn't make sense 🤔

Comment (ThibodeauJF) [reply packages/headless/src/features/sort-criteria/criteria.ts:134]: `The sort by "no sort" criterion.` ?

File: packages/headless/src/features/sort-criteria/criteria.ts [hunk 6]
```diff
@@ -1,22 +1,58 @@
...
+  /**
+   * Use standard index ranking factors (adjacency, TDIDF) and custom ranking expressions (QREs and QRFs) to compute a ranking score for each query result item, and sort the query results by descending score value
+   */
   Relevancy = 'relevancy',
+  /**
+   * Use only custom ranking expressions (QREs and QRFs) to compute a ranking score for each query result item, and sort the query results by descending score value.
+   */
   QRE = 'qre',
+  /**
+   * date ascending/date descending: Use the date field to sort the query results. This field typically contains the last modification date of each item.
```
Comment (jlblais) [inline packages/headless/src/features/sort-criteria/criteria.ts:24]: On line 28, you use `@[field]`. Should this be `@[date]`?  or vice versa?

**Review by jlblais — APPROVED** [review]

File: packages/headless/src/features/sort-criteria/criteria.ts (hunk 3 above)
Comment (olamothe) [reply packages/headless/src/features/sort-criteria/criteria.ts:67]: The first one is for "the param name".

So, it reads as:
a param named `criterion` of the type `{SortCriterion}` : The criterion to translate...

File: packages/headless/src/features/sort-criteria/criteria.ts (hunk 6 above)
Comment (olamothe) [reply packages/headless/src/features/sort-criteria/criteria.ts:24]: It's different in this case. 
`@[fieldname] ascending/descending` would be the expression sent to the API to sort by field
`date ascending/descending` would be the expression sent to the API to sort by date (no leading `@` character).

**Pull request merged**